app/handler/unsubscribe_encoder.py

```diff
@@ -25,52 +40,106 @@ class UnsubscribeLink:
 
 class UnsubscribeEncoder:
     @staticmethod
-    def encode(action: UnsubscribeAction, data: int) -> UnsubscribeLink:
+    def encode(
+        action: UnsubscribeAction, data: Union[int, UnsubscribeOriginalData]
+    ) -> UnsubscribeLink:
         if config.UNSUBSCRIBER:
             return UnsubscribeLink(UnsubscribeEncoder.encode_mailto(action, data), True)
         return UnsubscribeLink(UnsubscribeEncoder.encode_url(action, data), False)
 
-    @staticmethod
-    def encode_subject(action: UnsubscribeAction, data: int) -> str:
-        if action == UnsubscribeAction.DisableAlias:
-            return f"{data}="
-        if action == UnsubscribeAction.DisableContact:
-            return f"{data}_"
-        if action == UnsubscribeAction.UnsubscribeNewsletter:
-            return f"{data}*"
+    @classmethod
+    def encode_subject(
+        cls, action: UnsubscribeAction, data: Union[int, UnsubscribeOriginalData]
+    ) -> str:
+        if (
+            action != UnsubscribeAction.OriginalUnsubscribeMailto
+            and type(data) is not int
+        ):
+            raise ValueError(f"Data has to be an int for an action of type {action}")
+        if action == UnsubscribeAction.OriginalUnsubscribeMailto:
+            if type(data) is not UnsubscribeOriginalData:
+                raise ValueError(
+                    f"Data has to be an UnsubscribeOriginalData for an action of type {action}"
+                )
+            # Initial 0 is the version number. If we need to add support for extra use-cases we can bump up this number
+            data = (0, data.alias_id, data.recipient, data.subject)
+        payload = (action.value, data)
+        serialized_data = (
+            base64.urlsafe_b64encode(json.dumps(payload).encode("utf-8"))
+            .rstrip(b"=")
+            .decode("utf-8")
+        )
+        signed_data = cls._get_signer().sign(serialized_data).decode("utf-8")
+        return f"{UNSUB_PREFIX}.{signed_data}"
```

Comment:
can we run into the limit of email subject length?

Reply:
We potentially can. The limit set by the rfc is almost 1000 chars if I remember correctly but I'm sure some clients won't allow more than 256/512 chars. I'm trying to encode it as tight as I can.

Reply:
For now maybe we can have a `LOG.e` in case the encoded data is too long so we know when this issue will ever happen?

Reply:
Added